pkg/agent/route/route_linux.go

```diff
@@ -726,7 +726,9 @@ func (c *Client) Reconcile(podCIDRs []string, svcIPs map[string]bool) error {
 		}
 		// IPv6 doesn't support "on-link" route, routes to the peer IPv6 gateways need to
 		// be added separately. So don't delete such routes.
-		if desiredIPv6GWs.Has(route.Dst.IP.String()) {
+		// If WireGuard is enabled, the route to peer IPv6 gateway should be deleted since
```

Comment:
When it starts, if there is an IPv6 "onlink" route, it means it was normal mode and the remote Node still exists. I think it's more proper to "replace" the routes together when we call `InstallNodeRoute` for this Node, instead of removing "onlink" route here and the other route in `InstallNodeRoute`.
The issue is essentially we thought `route.Replace` can achieve route removal and creation, but it didn't work for wireguard case as the latter has only 1 route.
I think we could probably first check if the normal case really needs two routes. If yes, we should fix how "replace" is implemented. If no, we should just remove the "onlink" route.

Reply:
After some digging, I found that [kernel > 4.16](https://github.com/torvalds/linux/commit/fc1e64e1092f62290d59151d16f9de0210e303c8) supports adding IPv6 routes with the `onlink` flag as IPv4 does. Do you think we can remove the routes for the peer gateway and add routes to peer CIDRs with the `onlink` flag directly?

Reply:
The [current kernel version requirement](https://github.com/antrea-io/antrea/blob/main/docs/getting-started.md#open-vswitch) is >4.6. I think it's not worth to bump up the requirement to 4.16 just for this, given that it's easy to resolve with existing solution. Besides, Ubuntu 18.04 (<18.04.2) ships with 4.15, which might still be used.

Reply:
Thank you, @tnqn. I think we can remove routes for peer gateway if WireGuard is enabled in `AddRoutes` or create a gateway route for WireGuard to replace the one in normal mode. I am not sure which one is better. I initially thought that `AddRoutes` is responsible for adding or replacing routes. But we did need to delete the route for WireGuard.

Reply:
Not only WireGuard, there is also an issue when changing from encap to noEncap mode (in same subnet case).
I think we shouldn't add an unneeded route for this bug, and it doesn't fix encap -> noEncap (in different subnet case) anyway.
I feel the proper fix is to clean up possible stable routes installed by other modes when it installs routes for one mode. For example:
```
routeToNodeGwIPReplaced := false
routeToPodCIDRReplaced := false
for _, route := range routes {
	if err := netlink.RouteReplace(route); err != nil {
		return fmt.Errorf("failed to install route to peer %s (%s) with netlink. Route config: %s. Error: %v", nodeName, nodeIP, route.String(), err)
	}
	if route.Dst == podCIDR {
		routeToPodCIDRReplaced = true
	} else if route.Dst == ... {
		routeToNodeGwIPReplaced = true
	}
}
if !routeToNodeGwIPReplaced {
	if err := netlink.RouteDel(...); err != nil {
		...
	}
}
if !routeToPodCIDRReplaced {
	if err := netlink.RouteDel(...); err != nil {
		...
	}
}
```

Reply:
I've updated the code per discussion. But I did not remove the route to peer Pod CIDR when the route to peer Pod CIDR was not replaced. I am not confident about removing the route by just setting the Pod CIDR as the only parameter. Will it delete the route added manually in noEncap mode plus different subnet cases?